### PR TITLE
Issue #668 - Changes for using json-automationrelevance instead of json-rev

### PR DESCRIPTION
**Pull request by shubham-y**

### CODE REVIEW

**Comment by marco-c** [comment]
@shubham-y did you test the function like I described at https://github.com/mozilla/code-coverage/issues/668#issuecomment-782926358? Is it returning the same result before/after your change?

Note: there are some tests that need to be adjusted, you can look at the test logs by clicking on "Details" on the right of "Community-TC" and then opening the failing `Code Coverage Backend checks: unit tests` task and then clicking on "View Live Log".

**Comment by shubham-y** [comment]
@marco-c yes, the function is returning the same result before and after my change.
I did look at test logs as you mentioned but struggling to find a way to pass that test.

**Comment by marco-c** [comment]
Did you manage to run the test locally?

**Comment by shreeagt** [comment]
@marco-c I have finally managed to run the test locally and made a few changes in `mock_hgmo` as pointed out by you.
After changes now I am facing a `<ExceptionInfo KeyError('changesets',) tblen=3>` exception.

**Comment by shubham-y** [comment]
@marco-c I have finally managed to run the test locally and made a few changes in mock_hgmo as pointed out by you.
After changes now I am facing a <ExceptionInfo KeyError('changesets',) tblen=3> exception.

**Comment by marco-c** [comment]
> @marco-c I have finally managed to run the test locally and made a few changes in mock_hgmo as pointed out by you.
After changes now I am facing a <ExceptionInfo KeyError('changesets',) tblen=3> exception.

This means the mock response you added doesn't have a `changesets` key, that you are using in the code at https://github.com/mozilla/code-coverage/pull/927/files#diff-6f7fa9ada2bb8845e9f5b8b0e5932694e9e465825627142e760f0df43b7a31e7R34.
You should make sure the mock response is the same you get when you hit the actual json-automationrelevance API.

If you can't figure it out, please push your changes so I can see them in the PR, otherwise I can't help you further.

**Comment by marco-c** [comment]
@shubham-y are you still interested in working on this?

**Comment by shubham-y** [comment]
@marco-c Yes, I am still interested to work on this.

**Comment by shubham-y** [comment]
> > @marco-c I have finally managed to run the test locally and made a few changes in mock_hgmo as pointed out by you.
> > After changes now I am facing a <ExceptionInfo KeyError('changesets',) tblen=3> exception.
> 
> This means the mock response you added doesn't have a `changesets` key, that you are using in the code at https://github.com/mozilla/code-coverage/pull/927/files#diff-6f7fa9ada2bb8845e9f5b8b0e5932694e9e465825627142e760f0df43b7a31e7R34.
> You should make sure the mock response is the same you get when you hit the actual json-automationrelevance API.
> 
> If you can't figure it out, please push your changes so I can see them in the PR, otherwise I can't help you further.

@marco-c   I have pushed my changes, can you please help me further?

File: backend/tests/conftest.py [hunk 1]
```diff
@@ -185,7 +185,7 @@ def _test_pushes(request):
     with responses.RequestsMock(assert_all_requests_are_fired=False) as resps:
         resps.add_callback(
             responses.GET,
-            re.compile("https://hg.mozilla.org/(.+)/json-rev/(.+)"),
+            re.compile("https://hg.mozilla.org/(.+)/json-automationrelevance/(.+)"),
             callback=_test_rev,
```
Comment (marco-c) [inline backend/tests/conftest.py:189]: `_test_rev` is the function that builds the response, you should update it too.
It is currently returning a response in the format json-rev returns, instead it should return a response in the same format as json-automationrelevance

File: backend/code_coverage_backend/hgmo.py [hunk 2]
```diff
@@ -29,9 +31,12 @@ def hgmo_revision_details(repository, changeset):
     url = HGMO_REVISION_URL.format(repository=repository, revision=changeset)
     resp = requests.get(url)
     resp.raise_for_status()
-    data = resp.json()
-    assert "pushid" in data, "Missing pushid"
-    out = data["pushid"], data["date"][0]
+    if resp.json().get("changesets") is None:
```
Comment (marco-c) [inline backend/code_coverage_backend/hgmo.py:34]: `changesets` should always be present, so let's make it an assertion (`assert "changesets" in data, "Missing changesets"`).

File: backend/tests/conftest.py (hunk 1 above)
Comment (shubham-y) [reply backend/tests/conftest.py:189]: I have already made changes in `_test_rev` function
https://github.com/mozilla/code-coverage/pull/927/files/ab47697240fc5215f52bac28294404bf09e1f7b5#diff-aacea3e16f9fecd307f1a11567ba7a1abb8a6007266aec3b89dd65454130b76cR147
What other changes need to be done?

File: backend/code_coverage_backend/hgmo.py (hunk 2 above)
Comment (shubham-y) [reply backend/code_coverage_backend/hgmo.py]: Changes pushed https://github.com/mozilla/code-coverage/pull/927/commits/df0682d63e9e7eae64047f383007498169dc88e1
Is there a need to add the same assertion in `test_closest_report` function?

File: backend/tests/conftest.py (hunk 1 above)
Comment (marco-c) [reply backend/tests/conftest.py:189]: The response returned by `_test_rev` should be in the json-automationrelevance format (e.g. https://hg.mozilla.org/mozilla-central/json-automationrelevance/0058e174c5d0a4f751398b5aefcce42adc511535).
You are still returning the response in the json-rev format: `resp = {"pushid": int(revision[:3]), "date": [time.time(), 0]}`.

Comment (shubham-y) [reply backend/tests/conftest.py:189]: Changes pushed 7276fed
Are there any more fields to be added in response other than `pushid` and `date`?

Comment (marco-c) [reply backend/tests/conftest.py:189]: No, those should be enough

**Comment by marco-c** [comment]
Could you rebase on top of current master?

**Review by marco-c — APPROVED** [review]
Looks good, thanks!

**Pull request merged**